chart/kubeapps/templates/_helpers.tpl

```diff
@@ -160,3 +160,74 @@ Frontend service port number
 8080
 {{- end -}}
 {{- end -}}
+
+{{/*
+Return the proper Docker Image Registry Secret Names
+*/}}
+{{- define "kubeapps.imagePullSecrets" -}}
+{{/*
+Helm 2.11 supports the assignment of a value to a variable defined in a different scope,
+but Helm 2.9 and 2.10 does not support it, so we need to implement this if-else logic.
+Also, we can not use a single if because lazy evaluation is not an option
+*/}}
+{{- if .Values.global }}
+{{- if .Values.global.imagePullSecrets }}
+imagePullSecrets:
+{{- range .Values.global.imagePullSecrets }}
+  - name: {{ . }}
+{{- end }}
+{{- else if or .Values.frontend.image.pullSecrets .Values.apprepository.image.pullSecrets .Values.apprepository.syncImage.pullSecrets .Values.hooks.image.pullSecrets .Values.tillerProxy.image.pullSecrets .Values.chartsvc.image.pullSecrets .Values.dashboard.image.pullSecrets .Values.authProxy.image.pullSecrets }}
```

Comment:
I think we don't need to add support for a different `pullSecrets` per image if we don't have a specific use case for it. The code will get simplified a lot if we omit this part.